### PR TITLE
Bazel support for embedded ate-pairing lib

### PR DESCRIPTION
**Pull request by mobileink**
Initial commit adding Bazel support.

### CODE REVIEW

File: src/camlsnark_c/libsnark-caml/depends/ate-pairing/include/cybozu/benchmark.hpp [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	@author MITSUNARI Shigeo
 */
 #if defined(_MSC_VER) && (MSC_VER <= 1500)
-	#include <cybozu/inttype.hpp>
+	#include "cybozu/inttype.hpp"
```
Comment (mrmr1993) [inline src/camlsnark_c/libsnark-caml/depends/ate-pairing/include/cybozu/benchmark.hpp:8]: I oppose this change, see my comment [here](https://github.com/o1-labs/snarky/pull/524#issuecomment-681968731) for more context.

**Review by mrmr1993 — COMMENTED** [review]
Bazel changes look good, but blocking until C++ file changes are reverted.

File: src/camlsnark_c/libsnark-caml/depends/ate-pairing/include/cybozu/benchmark.hpp (hunk 1 above)
Comment (mobileink) [reply src/camlsnark_c/libsnark-caml/depends/ate-pairing/include/cybozu/benchmark.hpp:8]: No prob, will fix.

**Comment by mobileink** [comment]
Cancelling, C++ implementation no longer used.

**Pull request closed without merging**